### PR TITLE
KAFKA-15073 Close stale PRs [2/n]

### PR DESCRIPTION
**Pull request by mumrah**
As a follow-up of #13827, this patch will allow the stale Github Action to actually close old PRs.

### CODE REVIEW

File: .github/workflows/stale.yml
Comment (chia7712) [inline .github/workflows/stale.yml:55]: Should we add [close-pr-message](https://github.com/actions/stale#close-pr-message) to explain our policy?

**Review by chia7712 — COMMENTED** [review]
@mumrah nice patch! 1.1k pending PRs is a bit crazy ...

File: .github/workflows/stale.yml
Comment (mumrah) [reply .github/workflows/stale.yml:55]: Yes, good idea.

File: .github/workflows/stale.yml
Comment (jlprat) [inline .github/workflows/stale.yml]: I know it's probably not doable, but it would be great to close only the PRs where the contributor dropped the ball, and leave the ones waiting for us maintainers to review open.
With the closing message we leave the door open for the contributor to reopen, maybe we can be more explicit stating that it's totally fine to reopen if the PR is under this case

**Review by jlprat — COMMENTED** [review]
I left a comment, let me know what you think

File: .github/workflows/stale.yml
Comment (mumrah) [reply .github/workflows/stale.yml]: From the actions/stale docs

> It means that any updates made, or any comments added to the issues or to the pull requests will restart the counter of days before marking as stale.

So, if the contributor adds a comment, it will reset the counter for being stale. I'll clarify this in the stale message.

There is probably some additional automation we can look into for raising our attention to do reviews.

**Review by jlprat — APPROVED** [review]
LGTM, I think leaving 30 days for the contributor to act is an acceptable amount of time. Also they can reopen if they feel the closing was wrong

**Comment by chia7712** [comment]
>  I think leaving 30 days for the contributor to act is an acceptable amount of time. Also they can reopen if they feel the closing was wrong

If we want to address "30 days", the value of "days-before-close" should be "30" rather than "120". see docs: https://github.com/actions/stale?tab=readme-ov-file#days-before-close
```
Since adding the stale label will alter the last update date, we can calculate the number of days from this date.
```

**Comment by mumrah** [comment]
Good catch, @chia7712 👍 -- will adjust accordingly.

**Review by chia7712 — APPROVED** [review]

**Comment by jlprat** [comment]
> >  I think leaving 30 days for the contributor to act is an acceptable amount of time. Also they can reopen if they feel the closing was wrong
> 
> If we want to address "30 days", the value of "days-before-close" should be "30" rather than "120". see docs: https://github.com/actions/stale?tab=readme-ov-file#days-before-close
> ```
> Since adding the stale label will alter the last update date, we can calculate the number of days from this date.
> ```
> 
> 

Great catch! I understood it as total time. But you are right adding a comment and label will modify the last state for the PR

**Review by jlprat — COMMENTED** [review]
Re LGTM for clarity.
Thanks @mumrah

**Comment by mumrah** [comment]
Looks like its working as expected.

I did a manual run here https://github.com/apache/kafka/actions/runs/10818124385/job/30013048044

The 25 oldest PRs are now labelled `stale` https://github.com/apache/kafka/pulls?q=is%3Aopen+label%3Astale+sort%3Acreated-asc

**Pull request merged**
